### PR TITLE
Modernize project structure and build system

### PR DESCRIPTION
**Pull request by SpotlightKid**
* docs: convert readme, changelog, authors, and install docs to Markdown
* feat: generate dev requirements via pip-tools
* feat: switch to meson / mesonpep517 build system
* feat: remove travis and Appveyor CI setup
* feat: add GitHub Actions workflow to build wheels with cibuildwheel

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    default_options: [
+        'warning_level=2'
+    ],
+    meson_version: '>=0.53.0'
+)
+
+# Dependencies
+cpp = meson.get_compiler('cpp')
+jack2_dep = dependency('jack', version:'>=1.9.11', required: false)
+jack1_dep = dependency('jack', version:'>=0.125.0, <1.0', required: false)
```
Comment (eli-schwartz) [inline meson.build:15]:
```suggestion
jack1_dep = dependency('jack', version:['>=0.125.0', '<1.0'], required: false)
```

Currently this incorrectly compares 1.9.x versions as valid jack1 found dependencies.

File: meson.build [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+alsa_dep = dependency('alsa', required: false)
+thread_dep = dependency('threads')
+coremidi_dep = dependency('threads', required: false)
+winmm_dep = dependency('threads', required: false)
```
Comment (eli-schwartz) [inline meson.build:19]: You already have this mandatory once... do you need to find it 3 times?

File: meson.build [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+subdir('rtmidi')
+
+if not get_option('wheel')
+    meson.add_install_script('meson_postinstall.py')
+endif
+
+summary({
+    'Debug messages (verbose)': get_option('verbose'),
+    'Build for wheel': get_option('wheel'),
+    'ALSA support': target_machine.system() == 'linux' and alsa_dep.found() and get_option('alsa'),
```
Comment (eli-schwartz) [inline meson.build:52]: Note target_machine is intended when doing a https://en.wikipedia.org/wiki/Cross_compiler#Canadian_Cross build of a compiler, and you want to specify the machine code format the cross-compiled binaries will do codegen for.

For most cases you want host_machine, which is the system that you're compiling for (either the current machine, or when cross compiling, the version you are cross compiling for).

File: meson.build [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+bindir = join_paths(prefix, get_option('bindir'))
+
+# Build jack_mix_box and generate _jack_mixer extension source
+subdir('src')
+
+# Build & install C extension module and Python package
+subdir('rtmidi')
+
+if not get_option('wheel')
+    meson.add_install_script('meson_postinstall.py')
```
Comment (eli-schwartz) [inline meson.build:46]: You should run this with the same python version you found in `pymod.find_installation()` as that may be a different python version than the one in $PATH as `python3`. So you want to make sure that it compiles the right format of bytecode.

**Review by eli-schwartz — CHANGES_REQUESTED** [review]

File: meson.build (hunk 1 above)
Comment (SpotlightKid) [reply meson.build]: Fixed in 53b6921.

File: meson.build (hunk 3 above)
Comment (SpotlightKid) [reply meson.build]: Fixed in aa28305.

File: meson.build (hunk 2 above)
Comment (SpotlightKid) [reply meson.build]: Those were just placeholders until I have figured / tried out the proper dependency checks for macOS and windows.

I've now put in checks for the CoreMIDI framework and `winmm` dependencies, put I'm sure I'll have to make more adjustments after the first CI runs.

File: meson.build (hunk 4 above)
Comment (SpotlightKid) [reply meson.build]: Fixed in dddbaeb80b5147187841caf6aa0ca395bbe520b5.

**Comment by mikacousin** [comment]
As you can see in https://github.com/flathub/com.github.mikacousin.olc/pull/10#issuecomment-1483685138, there is something broken with setuptools and flatpak.
This PR seems to solve the problem. Do you think it is possible to merge it?
Thank you for all your work.

File: .github/workflows/publish-to-test-pypi.yml [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+      run: pipx run twine check --strict dist/*
+
+    - uses: actions/upload-artifact@v3
+      with:
+        path: dist/*.tar.gz
+
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
```
Comment (Czaki) [inline .github/workflows/publish-to-test-pypi.yml:38]:
```suggestion
    strategy:
      fail-fast: false
```

To not cancel jobs, other jobs on the first fail. In final version could be removed but simplify debugging.

**Review by Czaki — COMMENTED** [review]
It looks like error is here:

```
c++.exe: error: unrecognized command-line option '-framework'
```

But the only place when I see passing this flag to the compilator is Macos builds.

**Pull request merged**